### DISCUSSION TITLE
Internationalized Routing - automatic locale detection over User's IP Address

### DISCUSSION THREAD

**Discussion by JohnDeved** [vercel/next.js#29746] [Ideas]
### Describe the feature you'd like to request

right now nextjs only redirects users to locale routes using the `Accept-Language` header, which is fine in most use cases.
The problem arises on webshops, since the geolocation is also very important to consider.
Many users don't have a country locale in their language header, causing visitors from Mexico to end up on the Spain website for example.

### Describe the solution you'd like

An opt-in setting to redirect users based on their IP address

### Describe alternatives you've considered

not sure what alternatives could be possible

**Comment by adamsoderstrom** [comment]
You could use the `geolocation` function from `@vercel/functions`. 
More about it here: https://vercel.com/kb/guide/geo-ip-headers-geolocation-vercel-functions
